samples/browser/src/androidTest/java/org/mozilla/samples/browser/rules/WebserverRule.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.browser.rules
+
+import android.os.Handler
+import android.os.Looper
+import androidx.test.platform.app.InstrumentationRegistry
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import java.io.IOException
+
+/**
+ * A [TestWatcher] junit rule that will serve content from assets in the test package.
+ */
+class WebserverRule : TestWatcher() {
```

Comment:
Could this rule be usable for other apps? If so we can put it in a shared component.

Reply:
Ha, yes, this could live in `support-android-test`. As part of working on this I thought we could come up with a instrumentation component that makes it nice to test apps that use components. For example code that does things like "Type something into the toolbar" is basically the same across all apps that use our toolbar - so we could provide a nice "top level" test API for that.

Reply:
💯 !